### PR TITLE
feat(segmented-button): Add new package for segmented button

### LINKED ISSUE #6069: Release MDC Segmented Button
Tracking bug to implement and release MDC Segmented Button (AKA Toggle Group Button).

- [x] Draft Design Doc
- [x] Get Design Doc reviewed
- [x] Build initial prototype
- [x] Create bare structure of Segmented button component
  - Create folder (`packages/mdc-segmented-button`)
  - Add bare minimum files including package.json, Sass files (bare minimum)
  - Add README file with bare minimum document structure
- [x] Add base styles and theme mixins
- [x] Add adapter interface & foundation (Add unit tests)
- [x] Implement MDC vanilla component - `component.ts` (Add unit tests)
- [x] Update README documentation in each above step.
- [x] Fully test (Manual) for accessibility.
  - Perform accessibility audit using chrome dev tool.
  - Manually test using screen readers (Minimum 2 screen readers)
- [x] Set appropriate visibility setting in NPM registry for `@material/segmented-button` package.
- [ ] UX review (Material Design)
- [ ] Remove package from [dependency test](https://github.com/material-components/material-components-web/blob/master/scripts/check-pkg-for-release.js) exemptions

### PR DESCRIPTION
**Pull request by gschrag**
create segmented button component to let user manipulate toggle values of a group of related buttons

### CODE REVIEW

File: packages/mdc-segmented-button/package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+    "name": "@material/segmented-button",
+    "version": "0.0.0",
+    "description": "The Material Components for the web segmented button component",
+    "license": "MIT",
```
Comment (abhiomkar) [inline packages/mdc-segmented-button/package.json:5]: Add `directory` field to this JSON file.

**Comment by abhiomkar** [comment]
Can you remove "Closes PR number" line from PR description. GitHub will automatically closes issue when this PR is merged which is not what we wanted.

**Comment by gschrag** [comment]
@abhiomkar do you know why adding the package.json "description" field in commit e76a6ad broke the dependency test?

**Comment by abhiomkar** [comment]
Test failure is expected. You'll need to edit CSS bundle factory to include new segmented styles. https://github.com/material-components/material-components-web/blob/master/scripts/webpack/css-bundle-factory.js


```js
'mdc.segmented-button': getAbsolutePath('/packages/mdc-segmented-button/styles.scss'),
```

**Comment by gschrag** [comment]
I've noticed that all other mdc components have a '<component name>.scss' (e.g. 'mdc-button.scss') instead of a 'styles.scss'. I've also noticed that this style decision and others discussed in go/material-sass-style don't match how the sass files are actually implemented by mdc components. Is go/material-sass-style meant for a different repository?

File: packages/material-components-web/material-components-web.scss [hunk 2]
```diff
@@ -43,6 +43,7 @@
 @use "@material/notched-outline/mdc-notched-outline";
 @use "@material/radio/mdc-radio";
 @use "@material/ripple/mdc-ripple";
+@use "@material/segmented-button/styles";
```
Comment (abhiomkar) [inline packages/material-components-web/material-components-web.scss:46]: Seems like dependency check does not like `/styles` pattern. Sending a fix now.

Comment (abhiomkar) [reply packages/material-components-web/material-components-web.scss:46]: Sent PR #6152 to fix CSS dependency test.

You'll have to rebase your branch after above PR is merged.

Comment (abhiomkar) [reply packages/material-components-web/material-components-web.scss:46]: PR #6152 is now merged. Rebase your branch to fix the failing test.

Comment (gschrag) [reply packages/material-components-web/material-components-web.scss:46]: Thanks for the patch!

**Pull request merged**